src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE_ARRAY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[])new Object[SIZE_ARRAY];
+    }
+
     @Override
     public void add(T value) {
-
+        growArray();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            size--;
```

Comment:
```suggestion
    return;
```
you added an element to the new index. why decrease size? after you added an element to the end of the array - just finish method execution